src/IDelegateRegistry.sol

```diff
@@ -10,40 +10,61 @@ pragma solidity ^0.8.20;
 interface IDelegateRegistry {
     /// @notice Delegation type
     enum DelegationType {
-        NONE,
         ALL,
         CONTRACT,
-        ERC721,
         ERC20,
+        ERC721,
         ERC1155
     }
 
-    /// @notice Info about a single delegation, used for onchain enumeration
-    struct DelegationInfo {
+    /// @notice Standardizes storage positions of delegation data
+    enum StoragePositions {
+        delegate,
+        vault,
+        rights,
+        contract_,
+        tokenId,
+        balance
+    }
+
+    /// @notice Struct for returning arbitrary delegations
+    struct Delegation {
         DelegationType type_;
+        bool enable;
+        address delegate;
         address vault;
+        bytes32 rights;
+        address contract_;
+        uint256 tokenId;
+        uint256 balance;
+    }
+
+    /// @notice Struct for batch delegations
+    struct BatchDelegation {
```

Comment:
Slightly slimmer to not have the vault address in here, less calldata in batchDelegate(), but does leave us with two annoying mostly parallel structs

Reply:
Honestly the additional call data doesn't bother me too much. More the potential discrepancy between `vault` in the calldata and `msg.sender` that is actually used